skimage/measure/fit.py

```diff
@@ -618,21 +613,14 @@ def _dynamic_max_trials(n_inliers, n_samples, min_samples, probability):
     if n_inliers == 0:
         return np.inf
 
-    nom = 1 - probability
-    if nom == 0:
+    if probability == 1:
         return np.inf
 
-    inlier_ratio = n_inliers / float(n_samples)
-    denom = 1 - inlier_ratio ** min_samples
-    if denom == 0:
+    if n_inliers == n_samples:
         return 1
-    elif denom == 1:
-        return np.inf
 
-    nom = np.log(nom)
-    denom = np.log(denom)
-    if denom == 0:
-        return 0
+    nom = np.log(1 - probability)
+    denom = np.log(1 - (n_inliers / n_samples) ** min_samples)
```

Comment:
use `math` instead of `numpy` for scalars

```suggestion
    nom = math.log(1 - probability)
    denom = math.log(1 - (n_inliers / n_samples) ** min_samples)
```